### PR TITLE
[Cosmos DB] Introduce new api version 2021-03-15

### PR DESCRIPTION
**Pull request by xujin-zhang**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changes in this PR
This PR introduces a new properties "defaultIdentity" into db account request/response, and this is the only change compared to the old 2021-01-15.

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [x] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet. 
Changes will be in Prod hopefully by mid of March.
 
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
Mid of March.

  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
N/A
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [x] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2021-03-15/examples/CosmosDBDatabaseAccountCreateMax.json [hunk 1]
```diff
@@ -0,0 +1,200 @@
+{
```
Comment (MehaKaushik) [inline specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2021-03-15/examples/CosmosDBDatabaseAccountCreateMax.json:1]: update this example

File: specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2021-03-15/examples/CosmosDBDatabaseAccountGet.json [hunk 2]
```diff
@@ -0,0 +1,95 @@
+{
```
Comment (MehaKaushik) [inline specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2021-03-15/examples/CosmosDBDatabaseAccountGet.json:1]: update this

File: specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2021-03-15/examples/CosmosDBDatabaseAccountList.json [hunk 3]
```diff
@@ -0,0 +1,88 @@
+{
```
Comment (MehaKaushik) [inline specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2021-03-15/examples/CosmosDBDatabaseAccountList.json:1]: this too

File: specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2021-03-15/examples/CosmosDBDatabaseAccountListByResourceGroup.json [hunk 4]
```diff
@@ -0,0 +1,73 @@
+{
```
Comment (MehaKaushik) [inline specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2021-03-15/examples/CosmosDBDatabaseAccountListByResourceGroup.json:1]: this too

File: specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2021-03-15/examples/CosmosDBDatabaseAccountPatch.json [hunk 5]
```diff
@@ -0,0 +1,169 @@
+{
```
Comment (MehaKaushik) [inline specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2021-03-15/examples/CosmosDBDatabaseAccountPatch.json:1]: this too

File: specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2021-03-15/examples/CosmosDBDatabaseAccountGet.json (hunk 2 above)
Comment (xujin-zhang) [reply specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2021-03-15/examples/CosmosDBDatabaseAccountGet.json:1]: Done, thank you!

File: specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2021-03-15/examples/CosmosDBDatabaseAccountList.json (hunk 3 above)
Comment (xujin-zhang) [reply specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2021-03-15/examples/CosmosDBDatabaseAccountList.json:1]: Done, thank you!

File: specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2021-03-15/examples/CosmosDBDatabaseAccountListByResourceGroup.json (hunk 4 above)
Comment (xujin-zhang) [reply specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2021-03-15/examples/CosmosDBDatabaseAccountListByResourceGroup.json:1]: Done, thank you!

File: specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2021-03-15/examples/CosmosDBDatabaseAccountCreateMax.json (hunk 1 above)
Comment (xujin-zhang) [reply specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2021-03-15/examples/CosmosDBDatabaseAccountCreateMax.json:1]: Done, thank you!

File: specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2021-03-15/examples/CosmosDBDatabaseAccountPatch.json (hunk 5 above)
Comment (xujin-zhang) [reply specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2021-03-15/examples/CosmosDBDatabaseAccountPatch.json:1]: Done, thank you!

File: specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2021-03-15/cosmos-db.json [hunk 6]
```diff
@@ -0,0 +1,8590 @@
...
+        "disableKeyBasedMetadataWriteAccess": {
+          "description": "Disable write operations on metadata resources (databases, containers, throughput) via account keys",
+          "type": "boolean"
+        },
+        "keyVaultKeyUri": {
+          "description": "The URI of the key vault",
+          "type": "string"
+        },
+        "defaultIdentity": {
+          "description": "The default identity for accessing key vault",
```
Comment (pilor) [inline specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2021-03-15/cosmos-db.json:6097]: This should be more descriptive about how it is expecting the identity to be specified.  I.e. a managed identity ID or an object id, etc...  Should also state how to specify a system assigned identity

File: specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2021-03-15/cosmos-db.json [hunk 7]
```diff
@@ -0,0 +1,8590 @@
...
+        "enableFreeTier": {
+          "description": "Flag to indicate whether Free Tier is enabled.",
+          "type": "boolean"
+        },
+        "apiProperties": {
+          "description": "API specific properties.",
+          "type": "object",
+          "$ref": "#/definitions/ApiProperties"
+        },
+        "enableAnalyticalStorage": {
```
Comment (pilor) [inline specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2021-03-15/cosmos-db.json:6113]: 2021-03-01-preview introduced new properties around restorable DBs that this API version does not have.  Is it intended to have a newer stable version that does not bring those forward?

File: specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2021-03-15/examples/CosmosDBDatabaseAccountCreateMax.json [hunk 8]
```diff
@@ -0,0 +1,205 @@
...
+            "isZoneRedundant": false
+          }
+        ],
+        "consistencyPolicy": {
+          "defaultConsistencyLevel": "BoundedStaleness",
+          "maxIntervalInSeconds": 10,
+          "maxStalenessPrefix": 200
+        },
+        "keyVaultKeyUri": "https://myKeyVault.vault.azure.net",
+        "defaultIdentity": "FirstPartyIdentity",
```
Comment (pilor) [inline specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2021-03-15/examples/CosmosDBDatabaseAccountCreateMax.json:53]: you support getting the KV key with your RPs first party identity?  How do you secure that so users can't point to a KV they don't have any access to which another customer may have given your first party app access to?

File: specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2021-03-15/cosmos-db.json (hunk 7 above)
Comment (MehaKaushik) [reply specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2021-03-15/cosmos-db.json:6113]: yes this is intended

File: specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2021-03-15/examples/CosmosDBDatabaseAccountCreateMax.json (hunk 8 above)
Comment (xujin-zhang) [reply specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2021-03-15/examples/CosmosDBDatabaseAccountCreateMax.json:53]: Hi, we always do "key vault access validation" to check if the new default identity to be changed into will have access to key vault or not. If NO, then a bad request will be thrown directly to the customer. 

When the db account is created, it's by default to use "FIrstPartyIdentity", which is an globally shared Service Principal, then later the users can choose to switch to a db account level identity like "SystemAssignedIdentity". Once the switch is done, then the db account will not count on the "FirstPartiyIdentity" any longer.

File: specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2021-03-15/cosmos-db.json (hunk 6 above)
Comment (xujin-zhang) [reply specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2021-03-15/cosmos-db.json:6097]: Done

**Comment by xujin-zhang** [comment]
There's a "Swagger LintDiff " check failure. The feailure will be gone if we add the 
"$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/systemData"
In the cosmos-db.json 
and related 5 examples. 
However, the previous 2021-01-15 successfully suppress the error, we're still seeing how to suppress such error.

File: specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2021-03-15/examples/CosmosDBDatabaseAccountCreateMax.json (hunk 8 above)
Comment (pilor) [reply specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2021-03-15/examples/CosmosDBDatabaseAccountCreateMax.json:53]: Please make sure this approach goes through some level of security reviews / threat modeling

**Comment by jianyexi** [comment]
@xujin-zhang  There are 3 lintDiff errors called  'RequiredReadonlySystemData' which are introduced by this PR, could you fix it ? 
https://github.com/Azure/azure-rest-api-specs/pull/13253#issuecomment-789413400

**Comment by xujin-zhang** [comment]
> @xujin-zhang There are 3 lintDiff errors called 'RequiredReadonlySystemData' which are introduced by this PR, could you fix it ?
> [#13253 (comment)](https://github.com/Azure/azure-rest-api-specs/pull/13253#issuecomment-789413400)

**Comment by xujin-zhang** [comment]
> @xujin-zhang There are 3 lintDiff errors called 'RequiredReadonlySystemData' which are introduced by this PR, could you fix it ?
> [#13253 (comment)](https://github.com/Azure/azure-rest-api-specs/pull/13253#issuecomment-789413400)

Hi, as we discussed in the email thread, we want to suppress that error because our RP doesn't support that systemData yet. Can I have your help to suppress the error and sign-off my PR? Thank you!

**Review by MehaKaushik — APPROVED** [review]

**Pull request merged**